src/components/Todos/index.tsx

```diff
@@ -0,0 +1,204 @@
+  onUpdate: (todo: Todo) => Promise<void>;
+  onLoading: (status: boolean) => void;
+  onError: (status: string) => void;
+  onIdTodo: (id: number) => void;
+  isSubmitting: boolean;
+  onSubmit: (updateTodo: Todo | null) => void;
+};
+
+export const Todos: React.FC<Props> = ({
+  todo,
+  idTodo,
+  onDelete,
+  onUpdate,
+  onLoading,
+  onError,
+  onIdTodo,
+  isSubmitting,
+  onSubmit,
+}) => {
+  const { title, userId, id, completed } = todo;
+  const [updateTodo, setUpdateTodo] = useState<Todo | null>(null);
+
+  const titleField = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (updateTodo) {
+      titleField.current?.focus();
+    }
+  }, [updateTodo]);
+
+  const handleChecked = async (event: ChangeEvent<HTMLInputElement>) => {
+    onLoading(true);
+    onIdTodo(id);
+    const updateCompletedTodo = {
+      ...todo,
+      completed: event.target.checked,
+    };
+
+    try {
+      await onUpdate(updateCompletedTodo);
+      setUpdateTodo(null);
+    } catch (error) {
+      onError(ErrorMessages.UNABLE_TO_UPDATE_TODO);
+    } finally {
+      onLoading(false);
+      onIdTodo(0);
+    }
+  };
+
+  const handleKeyUp = async (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setUpdateTodo(null);
+      onIdTodo(0);
+      onLoading(false);
+
+      return;
+    }
+
+    if (event.key === 'Enter' && !isSubmitting) {
+      onLoading(true);
+      if (updateTodo?.title.trim() === '') {
+        try {
+          await onDelete(id);
+          setUpdateTodo(null);
+          onIdTodo(0);
+        } catch (error) {
+          onError(ErrorMessages.UNABLE_TO_DELETE_TODO);
+        } finally {
+          onLoading(false);
+        }
+      } else {
+        try {
+          await onSubmit(updateTodo);
+          setUpdateTodo(null);
+          onIdTodo(0);
+        } catch (error) {
+          onError(ErrorMessages.UNABLE_TO_UPDATE_TODO);
+        } finally {
+          onLoading(false);
+        }
+      }
+    }
+  };
+
+  const handleChangeTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setUpdateTodo(currentTodo => {
+      if (!currentTodo) {
+        return null;
+      }
+
+      return {
+        ...currentTodo,
+        title: event.target.value,
+      };
+    });
+  };
+
+  const handleBlur = async () => {
+    if (!isSubmitting) {
+      onLoading(true);
+      if (updateTodo?.title.trim() === '') {
+        try {
+          await onDelete(id);
+          setUpdateTodo(null);
+          onIdTodo(0);
+        } catch (error) {
+          onError(ErrorMessages.UNABLE_TO_DELETE_TODO);
+        } finally {
+          onLoading(false);
+        }
+      } else {
+        try {
+          await onSubmit(updateTodo);
+          setUpdateTodo(null);
+          onIdTodo(0);
+        } catch (error) {
+          onError(ErrorMessages.UNABLE_TO_UPDATE_TODO);
+        } finally {
+          onLoading(false);
+        }
+      }
+    }
+  };
```

Comment:
Here is some repeating logic
You can extract common logic into a separate function to avoid code duplication.
```suggestion

const resetState = () => {
    setUpdateTodo(null);
    onIdTodo(0);
    onLoading(false);
 };
  
const handleError = (errorType) => {
    onError(errorType);
    resetState();
 };
  
const handleBlur = async () => {
  if (isSubmitting) return;

  onLoading(true);
  
  try {
    if (!updateTodo?.title.trim()) {
      await onDelete(id);
    } else {
      await onSubmit(updateTodo);
    }
    resetState();
  } catch (error) {
    handleError(
      !updateTodo?.title.trim()
        ? ErrorMessages.UNABLE_TO_DELETE_TODO
        : ErrorMessages.UNABLE_TO_UPDATE_TODO
    );
  }
};

 
```